diff --git a/node_modules/react-native-worklets/android/src/networking/com/swmansion/worklets/WorkletsModule.kt b/node_modules/react-native-worklets/android/src/networking/com/swmansion/worklets/WorkletsModule.kt
index 7e2346c..3832edf 100644
--- a/node_modules/react-native-worklets/android/src/networking/com/swmansion/worklets/WorkletsModule.kt
+++ b/node_modules/react-native-worklets/android/src/networking/com/swmansion/worklets/WorkletsModule.kt
@@ -14,7 +14,6 @@ import com.facebook.react.turbomodule.core.CallInvokerHolderImpl
 import com.facebook.soloader.SoLoader
 import com.swmansion.worklets.runloop.AnimationFrameCallback
 import com.swmansion.worklets.runloop.AnimationFrameQueue
-import java.util.concurrent.atomic.AtomicBoolean
 
 @Suppress("KotlinJniMissingFunction")
 @ReactModule(name = WorkletsModule.NAME)
@@ -50,7 +49,8 @@ class WorkletsModule(
      * Invalidating concurrently could be fatal. It shouldn't happen in a normal flow, but it doesn't
      * cost us much to add synchronization for extra safety.
      */
-    private val mInvalidated = AtomicBoolean(false)
+    private val mInvalidationLock = Any()
+    private var mInvalidated = false
 
     @OptIn(FrameworkAPI::class)
     private external fun initHybrid(
@@ -161,9 +161,17 @@ class WorkletsModule(
         return mSlowAnimationsEnabled
     }
 
+    override fun initialize() {
+        reactApplicationContext.addLifecycleEventListener(this)
+    }
+
     override fun invalidate() {
-        if (mInvalidated.getAndSet(true)) {
-            return
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mInvalidated = true
+            reactApplicationContext.removeLifecycleEventListener(this)
         }
         if (mHybridData != null && mHybridData!!.isValid) {
             invalidateCpp()
@@ -176,11 +184,21 @@ class WorkletsModule(
     private external fun startCpp()
 
     override fun onHostResume() {
-        mAnimationFrameQueue.resume()
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mAnimationFrameQueue.resume()
+        }
     }
 
     override fun onHostPause() {
-        mAnimationFrameQueue.pause()
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mAnimationFrameQueue.pause()
+        }
     }
 
     override fun onHostDestroy() {}
diff --git a/node_modules/react-native-worklets/android/src/no-networking/com/swmansion/worklets/WorkletsModule.kt b/node_modules/react-native-worklets/android/src/no-networking/com/swmansion/worklets/WorkletsModule.kt
index 56bf305..2b3fa75 100644
--- a/node_modules/react-native-worklets/android/src/no-networking/com/swmansion/worklets/WorkletsModule.kt
+++ b/node_modules/react-native-worklets/android/src/no-networking/com/swmansion/worklets/WorkletsModule.kt
@@ -11,7 +11,6 @@ import com.facebook.react.turbomodule.core.CallInvokerHolderImpl
 import com.facebook.soloader.SoLoader
 import com.swmansion.worklets.runloop.AnimationFrameCallback
 import com.swmansion.worklets.runloop.AnimationFrameQueue
-import java.util.concurrent.atomic.AtomicBoolean
 
 @Suppress("KotlinJniMissingFunction")
 @ReactModule(name = WorkletsModule.NAME)
@@ -46,7 +45,8 @@ class WorkletsModule(
      * Invalidating concurrently could be fatal. It shouldn't happen in a normal flow, but it doesn't
      * cost us much to add synchronization for extra safety.
      */
-    private val mInvalidated = AtomicBoolean(false)
+    private val mInvalidationLock = Any()
+    private var mInvalidated = false
 
     @OptIn(FrameworkAPI::class)
     private external fun initHybrid(
@@ -117,9 +117,17 @@ class WorkletsModule(
         return mSlowAnimationsEnabled
     }
 
+    override fun initialize() {
+        reactApplicationContext.addLifecycleEventListener(this)
+    }
+
     override fun invalidate() {
-        if (mInvalidated.getAndSet(true)) {
-            return
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mInvalidated = true
+            reactApplicationContext.removeLifecycleEventListener(this)
         }
         if (mHybridData != null && mHybridData!!.isValid) {
             invalidateCpp()
@@ -132,11 +140,21 @@ class WorkletsModule(
     private external fun startCpp()
 
     override fun onHostResume() {
-        mAnimationFrameQueue.resume()
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mAnimationFrameQueue.resume()
+        }
     }
 
     override fun onHostPause() {
-        mAnimationFrameQueue.pause()
+        synchronized(mInvalidationLock) {
+            if (mInvalidated) {
+                return
+            }
+            mAnimationFrameQueue.pause()
+        }
     }
 
     override fun onHostDestroy() {}
